src/gc/proxy.d

```diff
@@ -307,6 +311,13 @@ extern (C)
         return proxy.gc_runFinalizers( segment );
     }
 
+    bool gc_inFinalizer() nothrow
+    {
+        if( proxy is null )
```

Comment:
Not like in the code you just added e.g. in line 499. But I'll leave this for a later pass.